models/evc.py

```diff
@@ -1201,51 +1198,34 @@ def check_list_traces(list_circuits):
         if not list_circuits:
             return {}
         uni_list = []
-        circuit_data = {}
-        for circuit_id in list_circuits:
-            circuit = list_circuits[circuit_id]
+        for circuit in list_circuits:
             # if a inter-switch EVC does not have current_path, it does not
             # make sense to run sdntrace on it
             if not circuit.is_intra_switch() and not circuit.current_path:
+                list_circuits.remove(circuit)
                 continue
             uni_list.append(circuit.uni_a)
             uni_list.append(circuit.uni_z)
-            interface_a = uni_to_str(circuit.uni_a)
-            circuit_data[interface_a] = {
-                                            'circuit_id': circuit.id,
-                                            'trace_name': 'trace_a'
-                                        }
-            interface_z = uni_to_str(circuit.uni_z)
-            circuit_data[interface_z] = {
-                                            'circuit_id': circuit.id,
-                                            'trace_name': 'trace_z'
-                                        }
 
         traces = EVCDeploy.run_bulk_sdntraces(uni_list)
+        traces = traces["result"]
 
-        circuit_by_traces = {}
         circuits_checked = {}
 
-        for trace_switch in traces:
-            for trace in traces[trace_switch]:
-                id_trace = str(trace[0]['dpid']) + ':' + str(trace[0]['port'])
-                if 'vlan' in trace[0]:
-                    id_trace += ':' + str(trace[0]['vlan'])
-                circuit_from_data = circuit_data.get(id_trace)
-                if circuit_from_data is None:
+        try:
+            for i, circuit in enumerate(list_circuits):
+                trace_a = traces[2*i]
+                trace_z = traces[2*i+1]
+                if not (trace_a and trace_z):
```

Comment:
Gretel, for testability let's delegate this `if not (trace_a and trace_z)` as an early return into `EVCDeploy.check_trace(circuit, trace_a, trace_z)`. 

Also shouldn't this be `if not trace_a or not trace_z`? If you got an empty list for any of these then you already know that the ultimate result it's incomplete anyway right?

Reply:
@viniarck, I have modified it according to your recommendation, now the circuit will appear in `circuits_checked` even though at least one of the traces is empty.
The condition `not(A and B) = not A or Not B`, but it is true that the second way is more intuitive, so I changed.

Reply:
Right. Looks good. I'll close this thread, thanks, Gretel.